app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -104,6 +107,31 @@ public class LoyaltyCardViewActivity extends CatimaAppCompatActivity implements
     final private TaskHandler mTasks = new TaskHandler();
     Runnable barcodeImageGenerationFinishedCallback;
 
+
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        // Access the SharedPreferences to check if volume keys for navigation are enabled
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+        boolean isVolumeKeysEnabled = sharedPreferences.getBoolean(getString(R.string.settings_key_use_volume_keys), false);
+
+        if (isVolumeKeysEnabled) {
```

Comment:
No need to use SharedPreferences, just use settings. See settings.getKeepScreenOn() for example.